### PR TITLE
KAFKA-6167: Timestamp on streams directory contains a colon, which is an illegal character

### PR DESCRIPTION
**Pull request by mjsax**
- change segment delimiter to .
 - added upgrade path
 - added test for old and new upgrade path

### CODE REVIEW

**Comment by mjsax** [comment]
This should go into `1.0.1` release.

Call for review @guozhangwang @dguy @bbejeck

**Comment by asfgit** [comment]
FAILURE
 6695 tests run, 1 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 6695 tests run, 1 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 6695 tests run, 1 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7962 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7962 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8070 tests run, 5 skipped, 0 failed.
--none--

**Comment by mjsax** [comment]
Test failures unrelated.

Retest this please.

**Comment by asfgit** [comment]
FAILURE
 8070 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7962 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8070 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8070 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7962 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7962 tests run, 5 skipped, 1 failed.
--none--

**Comment by mjsax** [comment]
Test failures are unrelated.

Retest this please.

**Comment by asfgit** [comment]
SUCCESS 
 8070 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8070 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8070 tests run, 5 skipped, 0 failed.
--none--

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java [hunk 1]
```diff
@@ -300,17 +299,16 @@ public void shouldThrowExceptionSettingStateListenerNotInCreateState() {
 
     @Test
     public void testNumberDefaultMetrics() {
-        final KafkaStreams streams = createKafkaStreams();
+        props.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, "1");
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java]: Are these changes related to this PR? They don't seem to be

Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java]: No. That was the fix for the failing test we extracted into an own PR (this fix is also in another of my PRs -- will rebase and clean it up)

**Comment by mjsax** [comment]
Rebased this.

**Comment by asfgit** [comment]
SUCCESS 
 8070 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7962 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 7962 tests run, 5 skipped, 1 failed.
--none--

**Comment by mjsax** [comment]
Retest this please.

**Comment by guozhangwang** [comment]
LGTM! Thanks @mjsax

**Comment by asfgit** [comment]
SUCCESS 
 8071 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8071 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8071 tests run, 5 skipped, 0 failed.
--none--
